java/mx/cider/orchard/LruMap.java

```diff
@@ -0,0 +1,207 @@
+package mx.cider.orchard;
+
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.concurrent.locks.ReadWriteLock;
+
+/**
+ * A <code>java.util.Map</code> with a capacity expressed as <code>max_size</code>
```

Comment:
You might mention that LRU stands for least recently used. :-) It took me a moment to remember this when I saw the class name.